### PR TITLE
 Reduce VirtualMachineScaleSetsClient#List calls for Azure cloud provider

### PR DESCRIPTION
**Pull request by feiskyer**
**What this PR does / why we need it**:

For master nodes not managed by VMSS, current cloud provider would updateCaches each time when finding master nodes info. This could result in call limits of `VirtualMachineScaleSetsClient#List`. 

This PR adds a caches to those nodes which reduces the cache updating significantly.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Continue of #43287.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/providers/azure/azure_util_vmss.go [hunk 1]
```diff
@@ -76,15 +76,17 @@ type scaleSet struct {
 	cacheMutex sync.Mutex
 	// A local cache of scale sets. The key is scale set name and the value is a
 	// list of virtual machines belonging to the scale set.
-	cache map[string][]scaleSetVMInfo
+	cache                     map[string][]scaleSetVMInfo
```
Comment (andyzhangx) [inline pkg/cloudprovider/providers/azure/azure_util_vmss.go]: nit: too many spaces here

File: pkg/cloudprovider/providers/azure/azure_util_vmss.go [hunk 2]
```diff
@@ -179,6 +181,11 @@ func (ss *scaleSet) getCachedVirtualMachine(nodeName string) (scaleSetVMInfo, er
 		return vm, nil
 	}
 
+	// Known node not managed by scale sets.
+	if ss.availabilitySetNodesCache.Has(nodeName) {
+		return scaleSetVMInfo{}, cloudprovider.InstanceNotFound
```
Comment (andyzhangx) [inline pkg/cloudprovider/providers/azure/azure_util_vmss.go]: why return `InstanceNotFound` if hit the cache?

File: pkg/cloudprovider/providers/azure/azure_util_vmss.go (hunk 1 above)
Comment (feiskyer) [reply pkg/cloudprovider/providers/azure/azure_util_vmss.go]: I think gofmt indents it with next field `availabilitySetNodesCache `

File: pkg/cloudprovider/providers/azure/azure_util_vmss.go (hunk 2 above)
Comment (feiskyer) [reply pkg/cloudprovider/providers/azure/azure_util_vmss.go]: Hitting availabilitySetNodesCache means the instance is not managed by scale sets, so returns InstanceNotFound here. Callers of `getCachedVirtualMachine()` will also use this error message to identify whether the node existing or not.

**Pull request merged**